### PR TITLE
#1959. Test valid combinations of modifiers. Part 2

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

**Review by eernstg — COMMENTED** [review]
Looks very good! Just a couple of types.

File: LanguageFeatures/Class-modifiers/syntax_abstract_base_class_A03_t01.dart [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+base mixin class BaseMixinImplementsAbstractBaseClass implements AbstractBaseClass {}
+//                                                               ^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+abstract base mixin class AbstractBaseMixinImplementsAbstractBaseClass implements AbstractBaseClass {}
+//                                                                                ^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Class-modifiers/syntax_abstract_base_class_A03_t01.dart:57]: Thank you for noticing. Added

File: LanguageFeatures/Class-modifiers/syntax_abstract_base_class_A03_t02.dart [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+abstract final class AbstractFinalClassImplementsBaseClass
+    implements AbstractBaseClass {}
+
+base mixin class BaseMixinImplementsAbstractBaseClass
+    implements AbstractBaseClass {}
+
+abstract base mixin class AbstractBaseMixinImplementsAbstractBaseClass
+    implements AbstractBaseClass {}
+
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Class-modifiers/syntax_abstract_base_class_A03_t02.dart:34]: Added

File: LanguageFeatures/Class-modifiers/syntax_abstract_final_class_A05_t03.dart [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+/// @description Checks that it is a compile-time error to declare a
+/// `base mixin` on an `abstract final class` in the same library where it is
+/// defined
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=class-modifiers
+
+abstract final class AbstractFinalClass {}
+
+base mixin BaseMixinOnFinalClass on AbstractFinalClass {}
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Class-modifiers/syntax_abstract_final_class_A05_t03.dart:16]: I don't know for sure. Specification mentions `extends` only, not `on`

> When the lint is enabled, a lint warning is reported if a class is not annotated `@reopen` and it:
> 
> - extends a class marked `interface` or `final` and is not itself marked `interface` or `final`, or
> - extends a `sealed` class which itself transitively extends a class marked `interface` or `final`.

**Review by sgrekhov — COMMENTED** [review]
Tests updated. I'm only not sure if `base mixin` `on` a `final` class is a `reopen` situation or not

**Review by eernstg — APPROVED** [review]
Thanks! LGTM.

**Comment by eernstg** [comment]
I'm closing the thread about the reopening at an `on` type, whatever we will do about this will not happen in this PR. I did comment on it, in order to get it clarified a bit.

**Pull request merged**